chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -220,9 +234,55 @@ def backward_gpu(self, inputs, grad_outputs):
         return gx, None
 
 
+def _double_backward_softmax_cross_entropy(x, t, normalize, class_weight,
+                                           ignore_label, reduce):
+    if isinstance(t, variable.Variable):
+        t = t.data
+
+    _check_class_weight_option(class_weight)
+    _check_reduce_option(reduce)
+    if chainer.is_debug():
+        _check_input_values(x, t, ignore_label)
+
+    loss = -chainer.functions.log_softmax(x)
+
+    if class_weight is not None:
+        shape = [1 if d != 1 else -1 for d in six.moves.range(x.ndim)]
+        class_weight = chainer.functions.broadcast_to(
+            class_weight.reshape(shape), x.shape)
+        loss = loss * class_weight
+
+    in_use = (t != ignore_label).astype(x.dtype)
+
+    loss = chainer.functions.rollaxis(loss, 1, loss.ndim)
+    loss = chainer.functions.reshape(loss, (-1, loss.shape[-1]))
+
+    # As F.select_item does not allow the elements of t to be negative,
+    # we substitute ignore_label with 0, which is valid.
+    t = t.copy()
+    xp = chainer.cuda.get_array_module((x, t))
+    t = xp.clip(t, 0, loss.shape[1] - 1)
```

Comment:
You don't have to do copy by yourself because `clip` does make a new array. The following is much simpler.
Before:
```
t = t.copy()
xp = chainer.cuda.get_array_module((x, t))
t = xp.clip(t, 0, loss.shape[1] - 1)
```
After
```
t = t.clip(0, loss.shape[1] - 1)
```